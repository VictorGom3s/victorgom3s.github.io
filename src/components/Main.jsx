import React from "react";
import { css } from "@emotion/core";

import Container from "./Container";

import * as githubIcon from "../images/eva_github-outline.svg";
import * as freeCodeCampIcon from "../images/cib_freecodecamp.svg";
import * as emailIcon from "../images/ic_baseline-alternate-email.svg";
import * as illustration from "../images/undraw_programming_2svr.svg";
import { background, secondary, primary } from "../colors";

const Main = () => {
  return (
    <section id="main" css={sectionStyle}>
      <Container>
        <div className="main" css={mainStyle}>
          <div css={textWrapper}>
            <div className="textContainer" css={textContainerStyle}>
              <h4 css={h4Style}>
                Olá, meu nome é <b css={boldText}>Victor Gomes</b>
              </h4>
              <h1 css={h1Style}>
                Eu sou um <b>Desenvolvedor Web</b>
              </h1>
              <div className="buttons" css={buttons}>
                <button className="btn btn-primary">Quero Contratar</button>
                <button className="btn btn-secondary">
                  Preciso de um Site
                </button>
              </div>
            </div>
            <div className="illustration" css={illustrationStyle}>
              <img src={illustration} alt="Illustration" />
            </div>
          </div>

          <div className="icons" css={iconsGrid}>
            <a href="https://github.com/victorgom3s">
              <img src={githubIcon} alt="GitHub Icon" />
            </a>
            <a href="mailto: vsampaio@pm.me">
              <img src={emailIcon} alt="E-mail Icon" />
            </a>
            <a href="https://www.freecodecamp.org/victorgom3s">
              <img src={freeCodeCampIcon} alt="FreeCodeCamp Icon" />
            </a>
          </div>
        </div>
      </Container>
    </section>
  );
};

const buttons = css`
  display: flex;
  justify-content: space-between;
  @media (min-width: 900px) {
    justify-content: flex-start;
  }
  align-items: center;
  align-content: center;
  .btn-primary {
    margin: 30px 10px 0px 0px;
  }
  .btn-secondary {
    margin: 30px 0px 0px 0px;
  }
`;

const boldText = css`
  color: ${primary};
`;

const sectionStyle = css`
  background-color: ${background};
  height: calc(100vh - 120px);
`;

const mainStyle = css`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 100%;
  flex-grow: 1;
`;

const iconsGrid = css`
  margin: 0 auto;
  display: flex;
  a {
    margin: 0;
    padding: 0;
    img {
      margin: 30px 10px;
      transition: 0.2s;
      &:hover {
        scale: 1.1;
      }
    }
  }
`;

const textWrapper = css`
  flex-grow: 1;
  margin: 0 auto;
  margin-bottom: 30px;
  display: flex;
  justify-content: space-between;
  width: 100%;
`;

const illustrationStyle = css`
  display: none;
  position: relative;
  @media (min-width: 900px) {
    display: flex;
    width: 60%;
    align-items: center;
    img {
      position: absolute;
      right: -20px;
      width: 100%;
      margin-left: 15px;
    }
  }
`;

const textContainerStyle = css`
  color: ${secondary};
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  align-content: center;
  justify-content: center;
  height: 100%;
  @media (min-width: 900px) {
    width: 40%;
    z-index: 1;
  }
`;

const h4Style = css`
  font-weight: normal;
  margin: 5px 0;
  font-size: 1.125rem;
`;

const h1Style = css`
  font-weight: normal;
  margin: 5px 0;
  font-size: 2.25rem;
`;

export default Main;
