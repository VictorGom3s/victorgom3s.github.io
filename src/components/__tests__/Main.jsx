import React from "react";
import renderer from "react-test-renderer";

import Main from "../Main";

describe("<Main/>", () => {
  it("Render correctly", () => {
    const tree = renderer.create(<Main />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
