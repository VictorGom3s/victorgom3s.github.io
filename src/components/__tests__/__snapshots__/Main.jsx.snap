// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> Render correctly 1`] = `
<section
  css={
    {
      "map": undefined,
      "name": "ebns4v",
      "next": undefined,
      "styles": "
  background-color: #2F2519;
  height: calc(100vh - 120px);
",
      "toString": [Function],
    }
  }
  id="main"
>
  <div
    className="container"
    css={
      {
        "map": undefined,
        "name": "1wzd80i",
        "next": undefined,
        "styles": "
        max-width: calc(100vw - 30%);
        margin: 0 auto;
        color: white;
        height: 100%;
        @media (max-width: 1024px) {
          max-width: calc(100vw - 10%);
        }
      ",
        "toString": [Function],
      }
    }
  >
    <div
      className="main"
      css={
        {
          "map": undefined,
          "name": "1m0dqp6",
          "next": undefined,
          "styles": "
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 100%;
  flex-grow: 1;
",
          "toString": [Function],
        }
      }
    >
      <div
        css={
          {
            "map": undefined,
            "name": "1jhs4jv",
            "next": undefined,
            "styles": "
  flex-grow: 1;
  margin: 0 auto;
  margin-bottom: 30px;
  display: flex;
  justify-content: space-between;
  width: 100%;
",
            "toString": [Function],
          }
        }
      >
        <div
          className="textContainer"
          css={
            {
              "map": undefined,
              "name": "hrjkbi",
              "next": undefined,
              "styles": "
  color: #F0EFED;
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  align-content: center;
  justify-content: center;
  height: 100%;
  @media (min-width: 900px) {
    width: 45%;
    z-index: 1;
  }
",
              "toString": [Function],
            }
          }
        >
          <h4
            css={
              {
                "map": undefined,
                "name": "1fsliq8",
                "next": undefined,
                "styles": "
  font-weight: normal;
  margin: 5px 0;
  font-size: 1.125rem;
  @media (min-width: 1100px) {
    font-size: 1.9rem;
  }
  @media (min-width: 1600px) {
    font-size: 2.1rem;
  }
",
                "toString": [Function],
              }
            }
          >
            main.myNameIs
             
            <b
              css={
                {
                  "map": undefined,
                  "name": "f517g3",
                  "next": undefined,
                  "styles": "
  color: #FA7D09;
",
                  "toString": [Function],
                }
              }
            >
              Victor Gomes
            </b>
          </h4>
          <h1
            css={
              {
                "map": undefined,
                "name": "rjoxjd",
                "next": undefined,
                "styles": "
  font-weight: normal;
  margin: 5px 0;
  font-size: 2.25rem;
  @media (min-width: 1100px) {
    font-size: 3rem;
  }
  @media (min-width: 1600px) {
    font-size: 4rem;
  }
",
                "toString": [Function],
              }
            }
          >
            main.iAm
             
            <b>
              main.softwareDeveloper
            </b>
          </h1>
        </div>
        <div
          className="illustration"
          css={
            {
              "map": undefined,
              "name": "8hkn9p",
              "next": undefined,
              "styles": "
  display: none;
  position: relative;
  @media (min-width: 900px) {
    display: flex;
    width: 50%;
    align-items: center;
    img {
      transform: scale(1.025);
      position: absolute;
      right: -35px;
      width: 100%;
    }
  }

  @media (min-width: 1025px) {
    img {
      transform: scale(1.25);
      right: -75px;
    }
  }
",
              "toString": [Function],
            }
          }
        >
          <img
            alt="Illustration"
            src="test-file-stub"
          />
        </div>
      </div>
      <div
        className="icons"
        css={
          {
            "map": undefined,
            "name": "16riqf4",
            "next": undefined,
            "styles": "
  margin: 0 auto;
  display: flex;
  a {
    margin: 0;
    padding: 0;
    img {
      margin: 30px 10px;
      transition: 0.2s;
      &:hover {
        transform: scale(1.1);
      }
    }
  }
",
            "toString": [Function],
          }
        }
      >
        <a
          href="https://github.com/victorgom3s"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="GitHub Icon"
            src="test-file-stub"
          />
        </a>
        <a
          href="mailto: o7xitc6vw@mozmail.com"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="E-mail Icon"
            src="test-file-stub"
          />
        </a>
        <a
          href="https://www.freecodecamp.org/victorgom3s"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="FreeCodeCamp Icon"
            src="test-file-stub"
          />
        </a>
      </div>
    </div>
  </div>
</section>
`;
